tests/test_blank_footprint.py

```diff
@@ -0,0 +1,221 @@
+
+
+def get_sim_with_dummy_module_registered(tmpdir=None, run=True, data=None):
+    """Return an initialised simulation object with a Dummy Module registered. If the `data` argument is provided,
+    the parameter in HealthSystem that holds the data on consumables availability is over-written."""
+
+    class DummyModule(Module):
+        METADATA = {Metadata.DISEASE_MODULE}
+
+        def read_parameters(self, data_folder):
+            pass
+
+        def initialise_population(self, population):
+            pass
+
+        def initialise_simulation(self, sim):
+            pass
+
+    # Create simulation with the HealthSystem and DummyModule
+    if tmpdir is not None:
+        _log_config = {
+            'filename': 'tmp',
+            'directory': tmpdir,
+        }
+    else:
+        _log_config = None
+
+    start_date = Date(2010, 1, 1)
+    sim = Simulation(start_date=start_date, seed=0, log_config=_log_config)
+
+    sim.register(
+        demography.Demography(resourcefilepath=resourcefilepath),
+        healthsystem.HealthSystem(resourcefilepath=resourcefilepath),
+        DummyModule(),
+        # Disable sorting + checks to avoid error due to missing dependencies
+        sort_modules=False,
+        check_all_dependencies=False
+    )
+
+    if data is not None:
+        sim.modules['HealthSystem'].parameters['availability_estimates'] = data
+
+    sim.make_initial_population(n=100)
+
+    if run:
+        sim.simulate(end_date=start_date)
+
+    return sim
+
+
+def get_dummy_hsi_event_instance(module, facility_id=None, blank = False):
+    """Make an HSI Event that runs for person_id=0 in a particular facility_id and requests consumables,
+    and for which its parent is the identified module."""
+
+    _facility_level = find_level_of_facility_id(facility_id)
+
+    class HSI_Dummy_no_blank(HSI_Event, IndividualScopeEventMixin):
+        def __init__(self, module, person_id):
+            super().__init__(module, person_id=person_id)
+            self.TREATMENT_ID = 'Dummy'
+            self.ACCEPTED_FACILITY_LEVEL = _facility_level
+            self._facility_id = facility_id
+            self.blank = blank_footprint
+            if self.blank:
+                self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({})
+            else:
+                self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({'ConWithDCSA': 1})
+
+
+
+        def apply(self, person_id, squeeze_factor):
+            if squeeze_factor != np.inf:
+                # Check that this appointment is being run and run not with a squeeze_factor that signifies that a cadre
+                # is not at all available.
+                self.did_run = True
+
+
+    hsi_dummy = HSI_Dummy_no_blank(module=module, person_id=0)
+    hsi_dummy.initialise()
+    hsi_dummy.facility_info = module.sim.modules['HealthSystem']._facility_by_facility_id[facility_id]
+    return hsi_dummy
+
+
+def get_dummy_hsi_event_instance_blank_footprint(module, facility_id=None):
```

Comment:
did you mean to collapse these two functions (get_dummy_hsi_event_instance_blank_footprint and get_dummy_hsi_event_instance_no_blank) and use the 'blank' kwarg to swtich between them?